packages/graphql/src/schema/get-where-fields.ts

```diff
@@ -126,30 +124,25 @@ function getWhereFields({
             }
 
             if (["String", "ID"].includes(f.typeMeta.name)) {
-                const matchesSetting: boolean | undefined = features?.filters?.[f.typeMeta.name]?.MATCHES;
-                if (matchesSetting !== undefined) {
-                    if (matchesSetting === true) {
-                        res[`${f.fieldName}_MATCHES`] = { type: "String", directives: deprecatedDirectives };
-                    }
-                } else if (enableRegex) {
-                    // TODO: This is deprecated. To be removed in 4.0.0.
-                    res[`${f.fieldName}_MATCHES`] = { type: "String", directives: deprecatedDirectives };
-                }
-
-                const stringWhereOperators = ["_CONTAINS", "_STARTS_WITH", "_ENDS_WITH"];
+                const stringWhereOperators: { comparator: string; typeName: string }[] = [
+                    { comparator: "_CONTAINS", typeName: f.typeMeta.name },
+                    { comparator: "_STARTS_WITH", typeName: f.typeMeta.name },
+                    { comparator: "_ENDS_WITH", typeName: f.typeMeta.name },
+                ];
 
                 const stringWhereOperatorsNegate = ["_NOT_CONTAINS", "_NOT_STARTS_WITH", "_NOT_ENDS_WITH"];
 
-                Object.entries(features?.filters?.String || {}).forEach(([filter, enabled]) => {
-                    if (filter === "MATCHES") {
-                        return;
-                    }
+                Object.entries(features?.filters?.[f.typeMeta.name] || {}).forEach(([filter, enabled]) => {
```

Comment:
Also, I think we could really make use of a `features` helper here, as this seems like it is going to be the way forwards for features flag and we should make our lifes a bit easier on this I think.

I'm thinking something a bit like:
```ts
features(f.typeMeta.name).hasFilter("MATCHES");
```

This way we have a more unified way of accesing features that don't require concatenation of `.?` and default objects and we could encapsulate the logic of traversing the filters, avoiding this `forEach` on our schema generation logic.

Thoughts?